src/industry_cmd.cpp

```diff
@@ -1125,7 +1125,7 @@ static void ProduceIndustryGoods(Industry *i)
 	if ((i->counter & 0x3F) == 0) {
 		uint32 r;
 		uint num;
-		if (Chance16R(1, 14, r) && (num = indsp->number_of_sounds) != 0 && _settings_client.sound.ambient) {
+		if (Chance16R(1, 14, r) && (num = indsp->number_of_sounds) != 0 && _settings_client.sound.ambient && *i->last_month_production > 0) {
```

Comment:
No need to dereference `i`

Reply:
I'm fairly new to C and not very familiar with pointers. How would I do that? Without the & (`&& i->last_month_production > 0`) I get a warning message: `warning: ordered comparison of pointer with integer zero [-Wextra]`. I'm probably not doing the right thing.

Reply:
Oh my bad, `i->last_month_production` is an array (= a pointer) so by dereferencing you effectively access `i->last_month_production[0]`

Reply:
Right, I just tried different things until I found what worked! That seems like the correct explanation. Thanks!

Reply:
Really, don't use the * operator to get the first element of an array, use array indexing for that.
But I'd strongly prefer if the entire array was checked for any non-zero values. It's very much possible to have a NewGRF industry with multiple outputs, and only some of them are active depending on the inputs provided.

Reply:
As is probably obvious, I'm pretty new to C, and I didn't realise that it was an array - I'll fix and push that. Thanks for the tips!